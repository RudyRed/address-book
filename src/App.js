import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { Helmet, HelmetProvider } from 'react-helmet-async';
import { ThemeProvider } from '@material-ui/core/styles';
import { CssBaseline } from '@material-ui/core';

import baseTheme from './themes';
import Details from './pages/Details';
import Home from './pages/Home';
import NotFound from './pages/NotFound';

function App() {
  return (
    <Router>
      <CssBaseline />
      <ThemeProvider theme={baseTheme}>
        <HelmetProvider>
          <Helmet titleTemplate="%s - Address Book" />
          <Switch>
            <Route exact path="/" component={Home} />
            <Route path="/:personId" component={Details} />
            <Route component={NotFound} />
          </Switch>
        </HelmetProvider>
      </ThemeProvider>
    </Router>
  );
}

export default App;
