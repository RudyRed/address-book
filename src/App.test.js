import { render, screen, waitFor } from '@testing-library/react';

import App from './App';
import mockedGetUsers from './__mocks__/App-getUsers.mock';

global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve(mockedGetUsers),
  }),
);

beforeEach(() => {
  fetch.mockClear();
});

describe('App', () => {
  it('Should render', async () => {
    render(<App />);

    expect(await screen.findByText('Address Book')).toBeInTheDocument();
  });

  it('matches snapshot', async () => {
    const { container } = render(<App />);

    await waitFor(() => expect(container).toMatchSnapshot());
  });
});
