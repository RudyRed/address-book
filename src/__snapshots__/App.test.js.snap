// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<div>
  <div
    class="Home-root-7"
  >
    <div
      class="Layout-root-8"
    >
      <div
        class="Layout-header-9"
      >
        <h1
          class="MuiTypography-root Layout-title-10 MuiTypography-h1"
        >
          Address Book
        </h1>
      </div>
      <main
        class="Layout-flex-12"
      />
    </div>
  </div>
</div>
`;
